skimage/transform/_seam_carving.pyx

```diff
@@ -0,0 +1,226 @@
+    cdef Py_ssize_t rows = img.shape[0]
+    cdef Py_ssize_t channels = img.shape[2]
+    cdef Py_ssize_t r, c, ch, shift
+    cdef Py_ssize_t c_shift
+
+    for r in range(rows):
+        shift = 0
+        for c in range(cols):
+            shift += seam_map[r, c]
+            c_shift = c + shift
+            for ch in range(channels):
+                img[r, c, ch] = img[r, c_shift, ch]
+
+
+def _seam_carve_v(img, energy_map, iters, border):
+    """ Carve vertical seams off an image.
+
+    Carves out vertical seams from an image while using the given energy map to
+    decide the importance of each pixel.[1]_
+
+    Parameters
+    ----------
+    img : (M, N) or (M, N, 3) ndarray
+        Input image whose vertical seams are to be removed.
+    energy_map : (M, N) ndarray
+        Cost array denoting importance of each pixel. The algorithm will try to
+        retain high valued pixels.
+    iters : int
+        Number of vertical seams to be removed.
+    border : int, optional
+        The number of pixels in the right, left and bottom end of the image
+        to be excluded from being considered for a seam. This is important as
+        certain filters just ignore image boundaries and set them to `0`.
+        By default border is set to `1`.
+
+    Returns
+    -------
+    image : (M, N - iters, 3) ndarray of float
+        The cropped image with the vertical seams removed.
+
+    References
+    ----------
+    .. [1] Shai Avidan and Ariel Shamir
+           "Seam Carving for Content-Aware Image Resizing"
+           http://www.cs.jhu.edu/~misha/ReadingSeminar/Papers/Avidan07.pdf
+    """
+    # This reference has been kept to be used for the `np.argsort` call
+    last_row_obj = np.zeros(img.shape[1], dtype=np.float)
+
+    cdef cnp.double_t[::1] last_row = last_row_obj
+    cdef Py_ssize_t[::1] sorted_indices
+    cdef cnp.uint8_t[:, ::1] seam_map = np.zeros(img.shape[0:2],
+                                                 dtype=np.uint8)
+    cdef Py_ssize_t cols = img.shape[1]
+    cdef Py_ssize_t rows = img.shape[0]
+    cdef Py_ssize_t seams_left = iters
+    cdef Py_ssize_t seams_removed
+    cdef Py_ssize_t seam_idx
+    cdef Py_ssize_t[::1] seam_buffer = np.zeros(rows, dtype=np.int)
+
+    cdef cnp.double_t[:, :, ::1] image = img
+    cdef cnp.int8_t[:, ::1] track_img = np.zeros(img.shape[0:2], dtype=np.int8)
+    cdef cnp.double_t[:, ::1] cumulative_img = np.zeros(img.shape[0:2],
+                                                        dtype=np.float)
+    cdef cnp.double_t[:, :, ::1] energy_img
+
+    energy_map[:, 0:border] = DBL_MAX
+    energy_map[:, cols-border:cols] = DBL_MAX
+
+    # Filters often let the boundary be `0`. If the last row is of all same
+    # values, the least value in then penultimate row will give 3 minimum
+    # enries in the last row. Hence, two successive removals will always
+    # intersect as the 3 least seams will share the same pixels except
+    # they will differ in the last row.
```

Comment:
Can you clarify this paragraph? There's at least one typo (the least value in *then* penultimate row), and overall I don't really understand what you're saying.